heartbeat/nagios4

```diff
@@ -0,0 +1,572 @@
+    # exit immediately if configuration is not valid
+    # nagios_validate_all || exit $?
+    # test if nagios is running.
+    get_nagios_status
+    case "$?" in
+        0)
+            rc=$OCF_SUCCESS
+            ocf_log debug "monitor: Resource is running"
+            ;;
+        1)
+            rc=$OCF_NOT_RUNNING
+            ocf_log debug "monitor: Resource is not running"
+            ;;
+        *)
+            ocf_log err "monitor: Resource has failed"
+            exit $OCF_ERR_GENERIC
+    esac
+    return $rc
+}
+
+# WHAT: return TRUE if a process with given PID is running
+# NOTE: input process id
+#       output true or false 
+ProcessRunning() {
+	ocf_log debug  "start: calling ProcessRunning "
+	RunningPID=$1
+	# Use /proc if it looks like it's here...
+	if [ -d /proc -a -d /proc/1 ]; then
+		[ -d /proc/$RunningPID ]
+	else
+		# This assumes we're running as root...
+		kill -s 0 "$RunningPID" >/dev/null 2>&1
+	fi
+}
+
+silent_status() {
+	ocf_log debug  "start: calling silent_status "
+	local OCF_RESKEY_PidFile="/var/nagios/nagios.pid"
+	if [ -f $OCF_RESKEY_PidFile ]; then
+	    local PID=`cat $OCF_RESKEY_PidFile`
+	ocf_log debug  "start: calling silent_status in if pid=$PID  "
+            ProcessRunning `cat $OCF_RESKEY_PidFile`
+	else
+	ocf_log debug  "start: calling silent_status in ELSE  pid=$PID  "
+		: No pid file
+		false
+	fi
+}
+
+
+# WHAT: return running PIDs if any or return false
+# WHY:  
+# NOTE: return 1 or running PIDs numbers
+nagios_running_pid(){
+        RunningPID=`ps auxww | grep nagios | grep -v grep | grep ${OCF_RESKEY_config} | awk '{print $2}'`
+	if [ ${RunningPID}=""  ]; then
+	        ocf_log debug "Nagios is not Running."
+		return ${OCF_NOT_RUNNING}
+	fi
+	ocf_log debug "Nagios Running PID=${RunningPID}"
+	return ${RunningPID}
+}
+
+# WHAT: find nagios running status
+# WHY:  
+nagios_file_pid(){
+    local OCF_RESKEY_PidFile=/var/nagios/nagios.pid
+	if [ -f ${OCF_RESKEY_PidFile} ]; then
+		NagiosPID=`head -n 1 ${OCF_RESKEY_PidFile}`
+                ocf_log debug "Nagios PID=${NagiosPID}"
+	else
+	        ocf_log debug "No Nagios PID lock file found in ${OCF_RESKEY_PidFile}"
+		return ${OCF_NOT_RUNNING}
+	fi
+	return $OCF_SUCCESS
+}
+
+get_nagios_pid(){
+	 ocf_log debug "get_nagios_pid: OCF_RESKEY_PidFile=${OCF_RESKEY_PidFile}"
+	if [ -f ${OCF_RESKEY_PidFile} ]; then
+		NagiosPID=`head -n 1 ${OCF_RESKEY_PidFile}`
+                ocf_log debug "get_nagios_pid: Nagios PID=${NagiosPID}"
+	else
+		 ocf_log debug "get_nagios_pid: No Nagios PID lock file found in ${OCF_RESKEY_PidFile}"
+		return ${OCF_NOT_RUNNING}
+	fi
+	return $OCF_SUCCESS
+}
+get_nagios_status(){
+	get_nagios_pid
+	# find if PID in ${OCF_RESKEY_PidFile}"
+	ps -p ${NagiosPID} > /dev/null 2>&1
+	if [ $? = "0" ]; then
+		return ${OCF_SUCCESS}
+	else
+    		return ${OCF_ERR_GENERIC}
+	fi
+}
+
+kill_nagios_process(){
+	if [ nagios_monitor ]; then
+		 ocf_log debug "killing Nagios PID: ${NagiosPID}"
+		kill ${2} ${NagiosPID}
```

Comment:
it could be a failure if trying to kill a non-exist process pid,

`$ kill 34242`
`bash: kill: (34242) - No such process`

`$ echo $?`
`1`